src/pipelines.py

```diff
@@ -183,39 +191,39 @@ def preprocessing_inference(config, model_name='unet'):
                                          },
                                 cache_dirpath=config.env.cache_dirpath)
 
-        loader = Step(name='loader',
-                      transformer=ImageSegmentationLoader(**config.loader),
-                      input_data=['specs'],
-                      input_steps=[reader_inference],
-                      adapter={'X': ([('reader_inference', 'X')]),
-                               'y': ([('reader_inference', 'y')]),
-                               'train_mode': ([('specs', 'train_mode')]),
-                               },
-                      cache_dirpath=config.env.cache_dirpath)
+    elif config.loader.dataset_params.image_source == 'disk':
+        reader_inference = Step(name='xy_inference',
+                                transformer=XYSplit(**config.xy_splitter[model_name]),
+                                input_data=['specs'],
+                                adapter={'meta': ([('input', 'meta')]),
+                                         'train_mode': ([('specs', 'train_mode')])
+                                         },
+                                cache_dirpath=config.env.cache_dirpath)
     else:
-        xy_inference = Step(name='xy_inference',
-                            transformer=XYSplit(**config.xy_splitter[model_name]),
-                            input_data=['input', 'specs'],
-                            adapter={'meta': ([('input', 'meta')]),
-                                     'train_mode': ([('specs', 'train_mode')])
-                                     },
-                            cache_dirpath=config.env.cache_dirpath)
-
-        loader = Step(name='loader',
-                      transformer=MetadataImageSegmentationLoader(**config.loader),
-                      input_data=['specs'],
-                      input_steps=[xy_inference, xy_inference],
-                      adapter={'X': ([('xy_inference', 'X')], squeeze_inputs),
-                               'y': ([('xy_inference', 'y')], squeeze_inputs),
-                               'train_mode': ([('specs', 'train_mode')]),
-                               },
-                      cache_dirpath=config.env.cache_dirpath)
+        raise NotImplementedError
+
+    loader = Step(name='loader',
+                  transformer=Loader(**config.loader),
+                  input_data=['specs'],
+                  input_steps=[reader_inference],
+                  adapter={'X': ([(reader_inference.name, 'X')], squeeze_inputs_if_needed),
+                           'y': ([(reader_inference.name, 'y')], squeeze_inputs_if_needed),
+                           'train_mode': ([('specs', 'train_mode')]),
+                           },
+                  cache_dirpath=config.env.cache_dirpath)
     return loader
 
 
 def mask_postprocessing(model, config, save_output=False):
+    if config.execution.loader_mode == 'crop_and_pad':
+        resize_function = crop_image
```

Comment:
@apyskir again I would change this naming. Can we somehow reuse things because there seems to be some repetition (feels like it at least)

Reply:
@jakubczakon Repetition, you mean the same thing in mask_postprocessing, postprocessing_masks and postprocessing_boarders?

Reply:
@apyskir Yeah it seems so. Maybe some `suffix` that changes the name would suffice (no pun intended)

Reply:
@jakubczakon I'm afraid we just have 3 different postprocessing pipelines for masks, masks (not the same masks) and borders, so we have to leave as it is